spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerPersistentEntity.java

```diff
@@ -40,4 +42,11 @@
 	 * @return the column names.
 	 */
 	Iterable<String> columns();
+
+	/**
+	 * Gets the Key for an entity instance.
+	 * @param entity the entity whose Key ID will be returned.
+	 * @return
+	 */
+	Key getId(T entity);
```

Comment:
Creating a `Key` object from an actual instance is a converter concern, not one of the meta-model.

Reply:
Done, removed from this interface